### PR TITLE
Switch to using psutil.

### PR DESCRIPTION
**Pull request by clalancette**
Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

I'll have to switch the `.gitmodules` back to the appropriate commit once we merge in https://github.com/ros-infrastructure/ros2-cookbooks/pull/39 .  Will have to be merged simultaneously with https://github.com/ros-infrastructure/ros2-cookbooks/pull/39 and https://github.com/ros2/ros2cli/pull/687

### CODE REVIEW

**Comment by clalancette** [comment]
Try at CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15967/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10639/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/16290/)

File: .gitmodules [hunk 1]
```diff
@@ -19,4 +19,4 @@
 [submodule "windows_docker_resources/ros2-cookbooks"]
 	path = windows_docker_resources/ros2-cookbooks
 	url = git@github.com:ros-infrastructure/ros2-cookbooks
-	branch = latest
+	branch = clalancette/switch-ifcfg-to-psutil
```
Comment (nuclearsandwich) [inline .gitmodules:22]: Let's make sure that we restore this before actually merging the PR.

File: linux_docker_resources/Dockerfile [hunk 2]
```diff
@@ -110,9 +111,6 @@ RUN apt-get update && apt-get install --no-install-recommends -y libopencv-dev
 # Install console_bridge for class_loader et al.
 RUN apt-get update && apt-get install --no-install-recommends -y libconsole-bridge-dev
 
-# Install build dependencies for class_loader.
-RUN apt-get update && apt-get install --no-install-recommends -y libpoco-dev
-
```
Comment (nuclearsandwich) [inline linux_docker_resources/Dockerfile:115]: This should get split into a separate PR.

**Review by nuclearsandwich — APPROVED** [review]
I spied a cleanup commit that should get split out into a separate PR since it will be very confusing after a squash-merge. Other than that and the noted pre-merge updates to the chef config this is good!

File: linux_docker_resources/Dockerfile (hunk 2 above)
Comment (clalancette) [reply linux_docker_resources/Dockerfile:115]: All right, that's fine by me.  Just noticed it while doing this.  Will remove it here.

Comment (clalancette) [reply linux_docker_resources/Dockerfile]: Fixed in 3464ce1.

**Comment by clalancette** [comment]
Since there have been changes both here and in https://github.com/ros-infrastructure/ros2-cookbooks/pull/39, here is another try at CI:

* Rolling Linux [Build Status](http://ci.ros2.org/job/ci_linux/15973/)
* Rolling Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10644/)
* Rolling Windows [Build Status](http://ci.ros2.org/job/ci_windows/16295/)

And on Foxy (just to check that ifcfg is still being installed there):
* Foxy Linux [Build Status](http://ci.ros2.org/job/ci_linux/15974/)
* Foxy Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10645/)
* Foxy Windows [Build Status](http://ci.ros2.org/job/ci_windows/16296/)

**Comment by clalancette** [comment]
Arg, I forgot to update this branch with the latest hash from ros2-cookbooks.  With that done, here is another try:

* Rolling Linux [Build Status](http://ci.ros2.org/job/ci_linux/15985/)
* Rolling Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10655/)
* Rolling Windows [Build Status](http://ci.ros2.org/job/ci_windows/16307/)

* Foxy Linux [Build Status](http://ci.ros2.org/job/ci_linux/15986/)
* Foxy Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10656/)
* Foxy Windows [Build Status](http://ci.ros2.org/job/ci_windows/16308/)

**Comment by clalancette** [comment]
Confirmed that this works as advertised; Foxy installs `ifcfg`, while Rolling installs `psutil`.

**Comment by clalancette** [comment]
Now that https://github.com/ros-infrastructure/ros2-cookbooks/pull/39 has been approved and merged, I set the .gitmodules back to the correct branch, and updated the submodule hash appropriately.  I'll go ahead and merge this one in as well, then deploy it.
